### PR TITLE
fix: Use origin as provided in HeadSeo. origin would be different for Domain cases

### LINKED ISSUE #14936: [CAL-3661] Wrong canonical link in source for org events
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3661](https://linear.app/calcom/issue/CAL-3661/wrong-canonical-link-in-source-for-org-events)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes #14936 
Fixes CAL-3661

Fixes the origin for Cal.com used in SEO. It fixes two things automatically
1. Canonical
2. og:url

## Tests
1. Added various involved components' unit tests
2. Added an e2e as well for org and non-org scenario

## Screenshots for the branch

Profile Page - Team
image

Profile Page - User
image

Event Page(BookerSeo)
image

For a non-org event it is same as earlier
image

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?
- Simply view the source-code for Profile Pages and Event Type(Booker) pages for Team and User events in an org

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by hariombalhara** [comment]
* **#14935** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/14935?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hariombalhara and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/features/bookings/components/BookerSeo.test.tsx
Comment (hariombalhara) [inline packages/features/bookings/components/BookerSeo.test.tsx:1]: Verify that `HeadSeo` is given the correct origin prop

File: apps/web/modules/bookings/views/bookings-single-view.test.tsx
Comment (hariombalhara) [inline apps/web/modules/bookings/views/bookings-single-view.test.tsx:1]: Verify that `HeadSeo` is given the correct origin prop

File: apps/web/modules/users/views/users-public-view.test.tsx
Comment (hariombalhara) [inline apps/web/modules/users/views/users-public-view.test.tsx:1]: Verify that `HeadSeo` is given the correct origin prop

File: packages/ui/components/head-seo/head-seo.test.tsx
Comment (hariombalhara) [inline packages/ui/components/head-seo/head-seo.test.tsx:1]: Tests HeadSeo component itself with the new origin prop support

File: apps/web/pages/team/[slug].tsx
Comment (hariombalhara) [inline apps/web/pages/team/[slug].tsx:1]: Similar to other page routes, we could refactor this component out in modules/teams/views/team-single-view.tsx and then write tests for it.

File: apps/web/modules/test-setup.ts
[the remark this answers is not in the document]
Comment (hariombalhara) [reply apps/web/modules/test-setup.ts:1]: Nice catch @sean-brydon !! Fixed it.

**Review by sean-brydon — APPROVED** [review]
LGTM - tested and seems like its working! 

Nice job with the tests <3

**Pull request merged**